providers/salesforce/bulk-read_test.go

```diff
@@ -0,0 +1,117 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockserver"
+	"github.com/amp-labs/connectors/test/utils/testroutines"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+)
+
+func TestBulkRead(t *testing.T) { //nolint:funlen,gocognit,cyclop,maintidx
+	t.Parallel()
+
+	responseUnknownObject := testutils.DataFromFile(t, "unknown-object.json")
+	responseAccount := testutils.DataFromFile(t, "bulk/read-launch-job-account.json")
+
+	tests := []bulkReadTestCase{
+		{
+			Name:         "Read object must be included",
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:         "At least one field is requested",
+			Input:        common.ReadParams{ObjectName: "Orders"},
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingFields},
+		},
+		{
+			Name:         "Mime response header expected",
```

Comment:
I don't think we govern response headers.

Reply:
As I wrote above it is for guarding against regression. We expect our code to handle this execution path. However, it does look repetitive, I am not sure how that can be generalised effectively and be applied to some if not all deep connectors.

Also by looking on these tests you can conclude what the code is doing. Tests preserve this by being a live contract. They are easy to run and done on every CI. Of course when the underlying provider changes due to breaking API the tests would need to follow and adapt. But for that there should be module support and per module tests.